plugins/net_plugin/net_plugin.cpp

```diff
@@ -3330,22 +3404,31 @@ namespace eosio {
       }
 
       if( accepted ) {
-         boost::asio::post( my_impl->thread_pool.get_executor(), [dispatcher = my_impl->dispatcher.get(), cid=c->connection_id, blk_id, msg]() {
-            fc_dlog( logger, "accepted signed_block : #${n} ${id}...", ("n", msg->block_num())("id", blk_id.str().substr(8,16)) );
-            dispatcher->add_peer_block( blk_id, cid );
+         boost::asio::post( my_impl->thread_pool.get_executor(), [dispatcher = my_impl->dispatcher.get(), c, blk_id, blk_num]() {
+            fc_dlog( logger, "accepted signed_block : #${n} ${id}...", ("n", blk_num)("id", blk_id.str().substr(8,16)) );
+            dispatcher->add_peer_block( blk_id, c->connection_id );
+
+            while (true) { // schedule any possible previously unlinkable blocks
+               unlinkable_block_state prev_unlinkable = my_impl->dispatcher->get_possible_linkable_block(blk_id);
```

Comment:
In this while loop, `blk_id` is never updated.

Reply:
Right. We want to try any blocks with a `previous` equal to the just applied `block_id`. I updated the comment. Hopefully clearer now.